coral-schema/src/main/java/com/linkedin/coral/schema/avro/SchemaUtilities.java

```diff
@@ -540,7 +553,8 @@ private static Schema getUnionFieldSchema(@Nonnull Schema leftSchema, @Nonnull S
       return makeNullable(leftSchema);
     }
     if (isNullableType(leftSchema) || isNullableType(rightSchema)) {
-      return makeNullable(getUnionFieldSchema(makeNonNullable(leftSchema), makeNonNullable(rightSchema), strictMode));
+      return makeNullable(getUnionFieldSchema(makeNonNullable(leftSchema), makeNonNullable(rightSchema), strictMode),
+          isNullSecond(leftSchema));
```

Comment:
could you add a comment here suggesting that if leftSchema and rightSchema are nullable union types with different ordering, we are choosing the ordering from the leftSchema. It could serve useful during future refactoring, if and when needed.